### ISSUE TITLE
module Beaker.Integrations.Ecto is not loaded and could not be found

### ISSUE THREAD

**Issue by KamilLelonek** [hahuang65/beaker#6]
I get this issue during compilation after running `mix do ecto.create, ecto.migrate`:

```elixir
== Compilation error on file lib/periscope/repo.ex ==
** (CompileError) lib/periscope/repo.ex:3: module Beaker.Integrations.Ecto is not loaded and could not be found
    (elixir) expanding macro: Kernel.use/1
    lib/supplements/repo.ex:3: Supplements.Repo (module)
    (elixir) lib/kernel/parallel_compiler.ex:100: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/8
```

And my `lib/periscope/repo.ex` file is as follows:

```elixir
defmodule Periscope.Repo do
  use Ecto.Repo, otp_app: :periscope
  use Beaker.Integrations.Ecto
end

```

**Comment by hahuang65** [comment]
Could you show me your `mix.exs`?

**Comment by KamilLelonek** [comment]
```elixir
defmodule Periscope.Mixfile do
  use Mix.Project

  def project do
    [
      app:             :periscope,
      version:         "0.0.1",
      elixir:          "~> 1.0",
      elixirc_paths:   elixirc_paths(Mix.env),
      compilers:       [:phoenix] ++ Mix.compilers,
      build_embedded:  Mix.env == :prod,
      start_permanent: Mix.env == :prod,
      deps:            deps,
   ]
  end

  def application do
    [
      mod:          { Periscope, [] },
      applications: apps,
    ]
  end

  defp apps do
    ~w(
      phoenix
      phoenix_ecto
      postgrex
      phoenix_html
      cowboy
      logger
      beaker
    )a
  end

  defp elixirc_paths(:test), do: elixirc_paths([]) ++ ["test/support"]
  defp elixirc_paths(_),     do: ["lib", "web"]

  defp deps do
    [
      {:phoenix,             "~> 1.0"},
      {:phoenix_ecto,        "~> 2.0"},
      {:postgrex,            "~> 0.10"},
      {:phoenix_html,        "~> 2.0"},
      {:cowboy,              "~> 1.0"},
      {:beaker,              "~> 1.1.2"},
    ]
  end
end
```

**Comment by hahuang65** [comment]
Could you try explicitly including ecto in your mix deps and apps list and see if that will fix it? Clearly ecto is loaded but perhaps I'm looking for the wrong thing in beaker to load ecto.

**Comment by KamilLelonek** [comment]
Same thing.

The problem happens only when the project is not compiled. So when I remove `_build` directory and try to build it again, the compilation fails. 

However, when I remove `use Beaker.Integrations.Ecto` entry, compile the project from scratch, and add the entry again, everything is working.

**Comment by KamilLelonek** [comment]
What I see during compilation is:

```elixir
==> beaker
Compiled lib/beaker/integrations/ecto.ex
Compiled lib/beaker.ex
Compiled lib/beaker/gauge.ex
Compiled lib/beaker/integrations/phoenix.ex
Compiled lib/beaker/time.ex
Compiled lib/beaker/counter.ex
Compiled lib/beaker/time_series/aggregated.ex
Compiled lib/beaker/time_series.ex
Compiled lib/web/helper.ex
Compiled lib/beaker/time_series/aggregator.ex
Compiled lib/web/router.ex
Compiled lib/web/web.ex
Compiled lib/web/controllers/metrics_controller.ex
Compiled lib/web/views/layout_view.ex
Compiled lib/web/views/metrics_view.ex
Generated beaker app
```

and later on when I check:

```elixir
Ecto.Repo |> Code.ensure_loaded? |> IO.inspect
```

I get `true`. It looks like compilation happens before `Ector.Repo` is loaded so later on when it becomes available, `Beaker.Integrations.Ecto` is not created.

**Comment by hahuang65** [comment]
This should be resolved by #7 and will be released as version 1.1.3.
